controllers/converter/apis.go

```diff
@@ -618,3 +633,228 @@ func filterPrefixes(src map[string]string, filterPrefixes []string) map[string]s
 	}
 	return dst
 }
+
+func ConvertScrapeConfig(promscrapeConfig *v1alpha1.ScrapeConfig, conf *config.BaseOperatorConf) *v1beta1vm.VMScrapeConfig {
+	cs := &v1beta1vm.VMScrapeConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        promscrapeConfig.Name,
+			Namespace:   promscrapeConfig.Namespace,
+			Annotations: filterPrefixes(promscrapeConfig.Annotations, conf.FilterPrometheusConverterAnnotationPrefixes),
+			Labels:      filterPrefixes(promscrapeConfig.Labels, conf.FilterPrometheusConverterLabelPrefixes),
+		},
+		Spec: v1beta1vm.VMScrapeConfigSpec{
+			MetricsPath:          promscrapeConfig.Spec.MetricsPath,
+			HonorTimestamps:      promscrapeConfig.Spec.HonorTimestamps,
+			Params:               promscrapeConfig.Spec.Params,
+			Scheme:               promscrapeConfig.Spec.Scheme,
+			BasicAuth:            ConvertBasicAuth(promscrapeConfig.Spec.BasicAuth),
+			Authorization:        convertAuthorization(promscrapeConfig.Spec.Authorization, nil),
+			TLSConfig:            ConvertSafeTlsConfig(promscrapeConfig.Spec.TLSConfig),
+			MetricRelabelConfigs: ConvertRelabelConfig(promscrapeConfig.Spec.MetricRelabelConfigs),
+			RelabelConfigs:       ConvertRelabelConfig(promscrapeConfig.Spec.RelabelConfigs),
+		},
+	}
+	if promscrapeConfig.Spec.HonorLabels != nil {
+		cs.Spec.HonorLabels = *promscrapeConfig.Spec.HonorLabels
+	}
+	if promscrapeConfig.Spec.ProxyConfig != nil && promscrapeConfig.Spec.ProxyURL != nil {
+		cs.Spec.ProxyURL = promscrapeConfig.Spec.ProxyURL
+	}
+	if promscrapeConfig.Spec.SampleLimit != nil {
+		cs.Spec.SampleLimit = *promscrapeConfig.Spec.SampleLimit
+	}
+	if promscrapeConfig.Spec.EnableCompression != nil {
+		cs.Spec.VMScrapeParams = &v1beta1vm.VMScrapeParams{
+			DisableCompression: pointer.BoolPtr(!*promscrapeConfig.Spec.EnableCompression),
+		}
+	}
+	if promscrapeConfig.Spec.ScrapeInterval != nil {
+		cs.Spec.ScrapeInterval = string(*promscrapeConfig.Spec.ScrapeInterval)
+	}
+	if promscrapeConfig.Spec.ScrapeTimeout != nil {
+		cs.Spec.ScrapeTimeout = string(*promscrapeConfig.Spec.ScrapeTimeout)
+	}
+	for _, staticConf := range promscrapeConfig.Spec.StaticConfigs {
```

Comment:
Would it be easier to have json.Marshal(PromScrapeConfig), json.Unmarshal(&VMScrapeConfig).

I think, we should have almost the same align of fields.

Reply:
yes, updated, please check.